keystone_json_assignment/json.py

```diff
@@ -76,56 +78,68 @@ def _get_public_id(self, user_id):
             'local_id': user_id,
             'entity_type': 'user'})
 
+    def load_user_project_map(self):
+        mtime = datetime.datetime.utcfromtimestamp(
+            os.path.getmtime('/etc/keystone/user-project-map.json'))
+        if self.userprojectmap == {} or mtime > self.last_loaded:
+            with open('/etc/keystone/user-project-map.json', 'r') as f:
+                userprojectmap = yaml.load(f)
+                self.last_loaded = mtime
+
+            projectnames = set()
+            for projects in userprojectmap.values():
+                for project in projects:
+                    projectnames.add(project)
+
+            projectidcache = {}
```

Comment:
Do we really want to delete the projectid cache on reload? I would assume that we only 'add' projects or change user assignments. Hence we just need to add new projects and keep the existing ones cached

Reply:
You don't think projects will ever be deleted?

Reply:
I find project deletion extremely unlikely, and if so all assignments would be removed anyway beforehand.